### DISCUSSION TITLE
XJT Lite not working on TX16S

### DISCUSSION THREAD

**Discussion by jarroyo78** [EdgeTX/edgetx#4455] [General]
TX16S Mk2 ELRS with BetaFPV Micro to Nano bay adapter and the XJT LITE.

XJT module momentarily lights up when powering up or if I unplug and plug back.

On the MODEL SETUP tab I turned off the Internal RF and have the External RF on.
Within the External RF the mode is set to XJT and D8 in my case. Tried using the D16 just to test but no luck.

I have seen others post about the same issue with the Zorro but no mention of the issue with the TX16S.

**Comment by pfeerick** [comment]
XJT Lite is not the same as XJT. I don't think it will work with the TX16 for two reasons... 1) lite (ACCESS) modules use two extra pins which are not routed on the 5 pin JR connector, thus your adapter does not have them connected to anything 2) the TX16 doesn't support using ACCESS modules with it (it's a hardware limitation of all but Frsky ACCESS capable radios). Natively, that is... there is an externalaccessmod that can be done with JR Micro modules. i.e. you'll only get that module (the XJTlite) to work with Frsky radios that support the nano module form factor natively, as well as the RM Zorro.

**Reply by jarroyo78** [reply]
Is there any workaround I could try? Would the XJT work?

**Reply by pfeerick** [reply]
Plain XJT module, yes, as that is a fully JR Micro bay compatible module (I can also confirm from personal experience as I have both modules). 

@raphaelcoeffic XJT lite could probably be made to work with TX16 using the External Module setting and connecting to AUX1, no?

**Reply by jarroyo78** [reply]
Could the Micro to Nano adapter be rewired to work with the XJT Lite?

**Reply by pfeerick** [reply]
That I don't know... the pins are there, but I don't know what circuitry (if any) is needed for the nano/lite -> JR conversion for ACCESS.

**Reply by jarroyo78** [reply]
Hope this helps and/or inspires someone to find a solution. Thanks for your time.

**Reply by pfeerick** [reply]
btw, if you _were_ thinking of getting a XJT module, I would strongly recommend getting a multi-protocol module (MPM/MULTI) instead... as that talks the same Frsky protocols, as well as a lot more. If you have mixed ACCST versions/regions, it will also allow you to use those without needing to reflash the TX or RXs so they are on the same version.

**Reply by jarroyo78** [reply]
Is that the RM 4in1, you are mentioning?

**Reply by pfeerick** [reply]
There are several (in JR Micro and nano form factors)... iRangeX IRX4, Jumper JP4IN1, RM 4IN1... even TBS make one (TBS MPM). They are all based on the same project ->  https://github.com/pascallanger/DIY-Multiprotocol-TX-Module

The RM 4IN1 is one of the more flexible options... as it is in a nano form factor, but comes with the JR bay adapter (which is identical in function to the BetaFPV one, just without the box that goes into the module bay).

**Reply by jarroyo78** [reply]
Man, thanks for all the advice. I wish I'd know sooner. My reasoning was if I can't get it to fit the TX16S I'll put it directly on the Pocket. I can't get it to work on neither.

**Reply by pfeerick** [reply]
Are you running a recent nightly on the Pocket? As, in fact, the XJT lite works with the Pocket (although you can't bind with telemetry - ~~which may or may not be a bug~~ which is indeed a bug/regression on 2.9 behaviour). If it's not binding but the lights are on, your receiver and xjt lite are probably running different ACCST versions or regions.

**Reply by jarroyo78** [reply]
The pocket is ELRS too. The thing with the pocket is that I can't get the module in. I feel I might break it.

**Reply by jarroyo78** [reply]
Took the whole back plate of the module off. Finally got the module board plugged in. Did the binding with no issues. But if I put it together it will not fit. It's just absurd.

**Reply by pfeerick** [reply]
My Pocket is ELRS also ;) That doesn't make any difference to whether it will work or not AFAIK... it's more that the connections need to be made to the nano module connector and a UART free for use on the MCU. Check the pins on the pocket's module bay to make sure they are all straight and parallel. Also, note that you can only get the module onto the locking tabs when it's about 6-7mm from the locked position... i.e. only there's only a couple of mm clearance between the pins and the sockets on the module when getting it into place.

**Reply by pfeerick** [reply]
Maybe a _minor_ alignment issue? I have to say the nano module bay format doesn't seem to be as well thought out as the JR Micro bay... using Swiss machine pins and sockets and relying on the module bay and module to line up that good was not well thought out by FrSky IMO... I've bent the pins on another radio due to that nonsense. Should have been something more spring-loaded that allowed for a bit more play.

**Reply by jarroyo78** [reply]
Update. 

Definitely an alignment issue. I ended up using a small phillips head to bend the pins on the Pocket by one degree I'd say. After that I assembled the module back and was able to finally use it.

Thanks again for your time.

**Reply by pfeerick** [reply]
Nice. Happy flying! :)

**Comment by raphaelcoeffic** [comment]
> XJT lite could probably be made to work with TX16 using the External Module setting and connecting to AUX1, no?

Yes, you need to have a full inverted UART on PPM / HB pins, just like for external ACCESS. You might have to bypass the compilation restrictions on module sizes as well (see `MODULE_SIZE_STD` vs `MODULE_SIZE_SML`).
